### PR TITLE
🔧 Fix local error handler

### PR DESCRIPTION
**Pull request by macabeus**
This PR fixes https://github.com/elysiajs/elysia/issues/966, following what's expected from the doc.

**BUT** it's still WIP because I started to wonder how is really the _expected_ behavior.
I think there is something missing being defined, or even if the local error worked someday because I could find no test and there are too many missing pieces here.

### CODE REVIEW

File: src/dynamic-handle.ts [hunk 1]
```diff
@@ -51,12 +52,25 @@ export const createDynamicHandler = (app: AnyElysia) => {
...
-				redirect
+				redirect,
+				error,
 			}
 		) as unknown as Context & {
 			response: unknown
 		}
 
+		const getLocalError = (response: unknown, hookError: HookContainer[]): Result | undefined => {
+			if (response instanceof ElysiaCustomStatusResponse) {
```
Comment (macabeus) [inline src/dynamic-handle.ts:63]: Can we be sure that whenever `response` is an `ElysiaCustomStatusResponse` it means an error?

File: test/lifecycle/error.test.ts [hunk 2]
```diff
@@ -272,4 +272,84 @@ describe('error', () => {
 			somePretty: 'json'
 		})
 	})
+
+	describe('handle scoped error', async () => {
```
Comment (macabeus) [inline test/lifecycle/error.test.ts:276]: @SaltyAom Please, can you confirm if these tests define the expected behavior when handling locally the error?

Comment (macabeus) [inline test/lifecycle/error.test.ts:276]: @SaltyAom If an exception is thrown, how should be the behavior? Should be the same as calling `error`, or should it go directly to the global error handler (i.e., `onError`)?

Currently, is the latter: it goes directly to the `onError`.
BUT I think that the expected behavior is going to the local handler first.

File: src/dynamic-handle.ts (hunk 1 above)
Comment (SaltyAom) [reply src/dynamic-handle.ts:63]: yes, `ElysiaCustomStatusResponse` is only return from `Elysia.error`

**Comment by SaltyAom** [comment]
Hi, thanks for taking up on the issue.

The expected behavior is local -> global which should be consistent with other lifecycles.

So in #966, it should log `handled`.

**Comment by macabeus** [comment]
Closing this PR, since the current behavior is working as intended.

**Pull request closed without merging**